lldb/docs/resources/addinglanguagesupport.md

```diff
@@ -0,0 +1,95 @@
+# Adding Programming Language Support
+
+LLDB has been architected to make it straightforward to add support for a
+programming language. Only a small enum in core LLDB needs to be modified to
+make LLDB aware of a new programming language. Everything else can be supplied
+in derived classes that need not even be present in the core LLDB repository.
+This makes it convenient for developers adding language support either in
+branches or downstream repositories since it practically eliminates the
```

Comment:
I am not convinced of using branches in the repo.

Reply:
I've removed it and only mentioned downstream repos.